### PR TITLE
Add more accessible theme switcher

### PR DESCRIPTION
**Pull request by marekzelinka**
Changes the theme switcher to be more accessible (credits to: [1]) and tweak the focus styles a bit.

[1]: [A Theme Switcher](https://inclusive-components.design/a-theme-switcher)

### CODE REVIEW

File: docs/theme/theme.css [hunk 1]
```diff
@@ -10,80 +10,50 @@
...
-  .theme:hover {
-    opacity: 1;
-  }
-  .theme__item + .theme__item {
-    margin-left: -0.8rem;
-  }
-  .theme:hover .theme__item + .theme__item {
-    margin-left: 0.5rem;
-  }
+.theme-switch__button[aria-pressed=false] {
```
Comment (dosyago-coder-0) [inline docs/theme/theme.css]: I think the accessibility / aria labels are a good idea. 

But you've changed *a lot* of styles here that seem unrelated to your core purpose with these commits.

Maybe I'm missing something, but are all these deletes necessary? 

*(I'm not a maintainer but I have the distinct feeling your PR has a better chance of being merged if you keep it focused and minimal.)*

**Comment by dvdsgl** [comment]
image

This looks pretty ugly.

**Comment by dvdsgl** [comment]
Reopen if you can limit the changes just to accessibility, and not change the switcher visually.

Thank you for your work!
